File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/ApiResponseModels/ResponseCourseRegisterModel.cs
using KnowledgeSharingApi.Domains.Enums;
using KnowledgeSharingApi.Domains.Models.Entities.Views;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.ApiResponseModels
{
    public class ResponseCourseRegisterModel : ViewCourseRegister
    {
        public Guid? UserRelationId { get; set; } = null;

        public EUserRelationType? UserRelationType { get; set; } = null;
    }
}

File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/api-response-models/response-course-register-model.js

import ViewCourseRegister from "../views/view-course-register";

class ResponseCourseRegisterModel extends ViewCourseRegister{
    constructor() {
        super();
        this.UserRelationType = null;
        this.UserRelationId = null;
    }

    init() {
        return new ResponseCourseRegisterModel();
    }
}

export default ResponseCourseRegisterModel;
